packages/react-storage/src/components/StorageBrowser/views/LocationDetailView/Controls.tsx

```diff
@@ -98,16 +98,24 @@ export const LocationDetailViewControls = ({
 
   const onPaginateNext = () => {
     if (hasInvalidPrefix || !nextToken) return;
-
+    dispatchStoreAction({ type: 'RESET_LOCATION_ITEMS' });
```

Comment:
This fixes an issue where the selected items weren't being cleared on pagination